### PR TITLE
fallback to Math.random for key generation on firefox

### PR DESCRIPTION
**Pull request by trevj**
churn has some issues on Firefox. This is the easy fix, working around this issue:
https://github.com/uProxy/uproxy/issues/1593

[Review on Reviewable](https://reviewable.io/reviews/uproxy/uproxy-lib/185)

### CODE REVIEW

File: src/churn/churn.ts [hunk 1]
```diff
@@ -172,8 +172,14 @@ var log :logging.Log = new logging.Log('churn');
   };
 
   // Generates a key suitable for use with CaesarCipher, viz. 1-255.
-  var generateCaesarKey_ = () : number => {
-    return (random.randomUint32() % 254) + 1;
+  var generateCaesarKey_ = (): number => {
+    try {
+      return (random.randomUint32() % 254) + 1;
```
Comment (bemasc) [inline src/churn/churn.ts]: I think you actually want % 255 here.  (255 % 255 == 0)

**Comment by bemasc** [comment]
:+1:

**Comment by willscott** [comment]
Keys should never be math.random. that already got flagged as a security
issue in our first review.

**Comment by bemasc** [comment]
This is a Caesar Cipher.  Using math.random hardly seems like the main cryptographic vulnerability.

**Comment by trevj** [comment]
Right, calling this a "key" is generous -- it just ensures we don't have a single, easily blockable signature. A very slight advance over a fixed "key".

File: src/churn/churn.ts (hunk 1 above)
Comment (trevj) [inline src/churn/churn.ts]: but we want a number from 1-255, so generate a number between 0-254, then add 1

(right?)

Comment (willscott) [inline src/churn/churn.ts]: `random % 254` will give you between 0-253.

Comment (trevj) [inline src/churn/churn.ts]: Oops. Thanks!

**Pull request merged**